lib/que/worker_group.rb

```diff
@@ -0,0 +1,70 @@
+# frozen_string_literal: true
+
+require_relative "worker"
+
+module Que
+  class WorkerGroup
+    DEFAULT_STOP_TIMEOUT = 5 # seconds
+
+    def self.start(count, **kwargs)
+      Que.logger.info(msg: 'Starting workers', event: 'que.worker.start', worker_count: count)
+
+      workers = Array.new(count) { Worker.new(**kwargs) }
+      worker_threads = workers.map { |worker| Thread.new { worker.work_loop } }
+
+      new(workers, worker_threads)
+    end
+
+    def initialize(workers, worker_threads)
+      @workers = workers
+      @worker_threads = worker_threads
+    end
+
+    attr_reader :workers
+
+    def each(&block)
```

Comment:
I'm personally not a huge fan of interfaces like this that kinda pretend to be a enumerable, but with some extra magic methods. Given that we only use this in one place, to call `collect_metrics` on each worker, can we not just define that method here and keep the interface to just three methods?